src/System.Text.Primitives/System/Text/Encoding/Utf16/Utf16TextEncodingLE.cs

```diff
@@ -7,113 +7,145 @@ namespace System.Text.Utf16
 {
     class Utf16TextEncodingLE : TextEncoder
     {
-        public override bool TryEncodeFromUtf8(ReadOnlySpan<byte> utf8, Span<byte> buffer, out int bytesWritten)
+        #region Encoding Constants
+
+        // TODO: Some of these members are needed only in Utf16LittleEndianEncoder.
+        //       Should we add the usage of them to UnicodeCodePoint class and merge this class with it?
+        private const uint Utf16HighSurrogateFirstCodePoint = 0xD800;
+        private const uint Utf16HighSurrogateLastCodePoint = 0xDFFF;
+        private const uint Utf16LowSurrogateFirstCodePoint = 0xDC00;
+        private const uint Utf16LowSurrogateLastCodePoint = 0xDFFF;
+
+        private const uint Utf16SurrogateRangeStart = Utf16HighSurrogateFirstCodePoint;
+        private const uint Utf16SurrogateRangeEnd = Utf16LowSurrogateLastCodePoint;
+
+        // To get this to compile with dotnet cli, we need to temporarily un-binary the magic values
+        private const byte b0000_0111U = 7;
```

Comment:
I think C#7 supports binary literals you could use after we merge the tools: https://github.com/dotnet/roslyn/issues/215